### ISSUE TITLE
recordPath doesn't make any sense

### ISSUE THREAD

**Issue by SamDecrock** [bluenviron/mediamtx#3969]
### Which version are you using?

1.9.3

### Which operating system are you using?

Linux amd64 Docker

### Describe how to replicate the issue

I'm running the following docker image of MediaMTX in Kubernetes: `bluenviron/mediamtx:1.9.3`

The config is set to
```yaml
paths:
  all_others:
    record: yes
    recordPath: ./recordings/%Y-%m-%d/%path--%Y-%m-%d--%Hh%M
    recordSegmentDuration: 30m
    recordDeleteAfter: 0s
```

But when checking the files being created, I see
```
$ TZ=utc ls -l
total 68176
-rw-r--r-- 1 root root 13984850 Nov 21 19:34 mycamera--2024-11-21--19h04.mp4
-rw-r--r-- 1 root root 13977877 Nov 21 20:04 mycamera--2024-11-21--20h04.mp4
-rw-r--r-- 1 root root 13979416 Nov 21 20:34 mycamera--2024-11-21--21h04.mp4
-rw-r--r-- 1 root root 13961680 Nov 21 21:04 mycamera--2024-11-21--22h04.mp4
-rw-r--r-- 1 root root 13892649 Nov 21 21:34 mycamera--2024-11-21--23h04.mp4
```
The first filename makes sense. Recording started at 19h04 and took 30 minutes,  hense the modification date of 19:34. But then the follow up filenames don't make any sense anymore. Especially if you look at the last one. It's like it's from the future 😮

**Comment by KosrenDQ** [comment]
I think the problem is the `recordSegmentDurction: 30m` configuration. MediaMTX will create segments that are at least 30 minutes long. As the example configuration suggests, try a lower value, such as `1s. '

**Comment by aler9** [comment]
Thanks for reporting the issue, this is fixed by #4062
